main.tf

```
@@ -443,7 +443,7 @@ resource "aws_subnet" "elasticache" {
 resource "aws_elasticache_subnet_group" "elasticache" {
   count = var.create_vpc && length(var.elasticache_subnets) > 0 && var.create_elasticache_subnet_group ? 1 : 0
 
-  name        = var.name
+  name        = var.elasticache_subnet_group_name != null ? lower(var.elasticache_subnet_group_name) : lower(var.name)
```

ya, not sure why some have `lower()` and some don't, but lets maintain status quo for now:
```suggestion
  name        = coalesce(var.elasticache_subnet_group_name, var.name)
```